src/main/java/io/tiledb/spark/TileDBDataReaderPartitionScan.java

```diff
@@ -619,35 +649,40 @@ private int getScalarValueColumn(String name, Datatype dataType, int index) thro
           }
           break;
         }
-      case TILEDB_DATETIME_MS:
-        {
-          long[] buff = (long[]) query.getBuffer(name);
-          bufferLength = buff.length;
-          numValues = bufferLength;
-          if (resultVectors.length > 0) {
-            resultVectors[index].reset();
-            resultVectors[index].putLongs(0, bufferLength, buff, 0);
-          }
-          break;
-        }
       default:
         {
           throw new TileDBError("Not supported getDomain getType " + dataType);
         }
     }
+    return numValues;
+  }
+
+  private int getScalarValueColumn(String name, Datatype dataType, int index) throws TileDBError {
+
+    metricsUpdater.startTimer(queryGetScalarAttributeTimerName);
+    boolean nullable = false;
+    int numValues;
+    short[] validityByteMap;
+
+    if (this.arraySchema.hasAttribute(name)) {
+      try (Attribute attr = this.arraySchema.getAttribute(name)) {
+        nullable = attr.getNullable();
+      }
+    }
+    numValues = putNonNullValuesOfNullableAttr(name, dataType, index);
```

Comment:
Should this be refactored? To my understanding, we process the whole batch twice, first when we call the `putNonNullValuesOfNullableAttr` and then when we look for null values in lines 675 and 676. As a result, for a batch of size `n` we end-up with a complexity of `2*n` as we iterate through all rows twice.

Reply:
As we discussed, it is not obvious which way is better. The code is simpler as it is now.